libyul/optimiser/FunctionCallFinder.cpp

```diff
@@ -16,23 +16,46 @@
 */
 
 #include <libyul/optimiser/FunctionCallFinder.h>
+
+#include <libyul/optimiser/ASTWalker.h>
 #include <libyul/AST.h>
 
 using namespace solidity;
 using namespace solidity::yul;
 
-std::vector<FunctionCall*> FunctionCallFinder::run(Block& _block, YulString _functionName)
+namespace
+{
+template<typename Base, typename ResultType>
+class MaybeConstFunctionCallFinder: Base
```

Comment:
Wouldn't it make sense to put this somewhere where this could be potentially reused? I mean it's a nice finder that could also be used to find other things like `FunctionCalls`, right?

Reply:
ok, after really reading the code it would need to get generalised a bit more. maybe its not worth it.

Reply:
In its current state it can already find anything that has a corresponding call operator in astmodifier/astwalker (void operator()(ResultType& _functionCall) override). so yeah, could pull this out into a separate header so that it can more easily be reused or we do that once the need arises - although then we'd have to remember that this exists.

Reply:
Yep, maybe it could make sense to generalise the condition function? I tried to create an example in the other comment. Of course that only would make sense if we need to collect different things.